apisix/healthcheck_manager.lua

```diff
@@ -0,0 +1,322 @@
+                    " current version: ", new_version, " requested version: ", resource_ver)
+            if resource_ver ~= new_version then
+                goto continue
+            end
+
+            -- if a checker exists then delete it before creating a new one
+            local existing_checker = working_pool[resource_path]
+            if existing_checker then
+                existing_checker.checker:delayed_clear(10)
+                existing_checker.checker:stop()
+                core.log.info("releasing existing checker: ", tostring(existing_checker.checker),
+                              " for resource: ", resource_path, " and version: ",
+                              existing_checker.version)
+            end
+            local checker = create_checker(upstream)
+            if not checker then
+                goto continue
+            end
+            core.log.info("create new checker: ", tostring(checker), " for resource: ",
+                        resource_path, " and version: ", resource_ver)
+            add_working_pool(resource_path, resource_ver, checker)
+        end
+
+        ::continue::
+        waiting_pool[resource_path] = nil
+    end
+end
+
+
+function _M.timer_working_pool_check()
+    if core.table.nkeys(working_pool) == 0 then
+        return
+    end
+
+    local working_snapshot = tab_clone(working_pool)
+    for resource_path, item in pairs(working_snapshot) do
+        --- remove from working pool if resource doesn't exist
+        local res_conf = fetch_latest_conf(resource_path)
+        local need_free = true
+        if res_conf and res_conf.value then
+            local current_ver = _M.upstream_version(res_conf.modifiedIndex,
+                                                    res_conf.value._nodes_ver)
+            core.log.info("checking working pool for resource: ", resource_path,
+                        " current version: ", current_ver, " item version: ", item.version)
+            if item.version == current_ver then
+                need_free = false
+            end
+        end
+
+        if need_free then
+            working_pool[resource_path] = nil
+            item.checker.dead = true
+            item.checker:delayed_clear(DELAYED_CLEAR_TIMEOUT)
+            item.checker:stop()
+            core.log.info("try to release checker: ", tostring(item.checker), " for resource: ",
+                        resource_path, " and version : ", item.version)
+        end
+    end
+end
+
+function _M.init_worker()
+    local timer_create_checker_running = false
+    local timer_working_pool_check_running = false
+    timer_every(1, function ()
+        if not exiting() then
+            if timer_create_checker_running then
+                core.log.warn("timer_create_checker is already running, skipping this iteration")
+                return
+            end
+            timer_create_checker_running = true
+            local ok, err = pcall(_M.timer_create_checker)
```

Comment:
`_M.timer_create_checker` vs `local function timer_create_checker`

I think we do not need to export `timer_create_checker`, local function should be enough